### PR TITLE
[ascend] update attn op_backend

### PR DESCRIPTION
**Pull request by wanfengcxz**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

DLINFER has updated the Ascend attention implementation from the legacy ATB
path to FIA/FIA v2 and unified graph replay with `torch.npu.NPUGraph.update()`.

This PR adds the corresponding LMDeploy-side adaptations.
Dlinfer pr: https://github.com/DeepLink-org/dlinfer/pull/350

## Changes

- Cache and reuse a fixed FIA causal mask per NPU device to avoid allocating
  the attention mask on every execution.
- Use the fixed split-fuse causal mask for Dense, GQA, MLA and paged-prefill
  attention with `sparse_mode=3`.
- Detect MLA models through `k_head_dim != v_head_dim` and explicitly pass
  `is_mla` to the DLINFER Ascend graph runner.
- Validate the torch/torch-npu runtime when initializing the Ascend backend.

The companion DLINFER update includes:

- Migrating prefill and paged attention to FIA/FIA v2.
- Adding MLA paged-prefill support.
- Removing unnecessary `.tolist()` calls from attention hot paths.
- Removing the legacy ATB graph-task attention path.
- Unifying graph replay through `NPUGraph.update()`.

## Dependency

This PR depends on the corresponding DLINFER attention update. The DLINFER PR
should be merged first, or both PRs should reference each other.

Required runtime:

- `torch >= 2.8.0`
- `torch-npu >= 2.8.0.post1`

## Test

- DeepSeek-V2-Lite MLA graph capture/replay
- Qwen3.5 Dense graph capture/replay
- Dense and MLA prefill/decode attention tests
- MLA paged-prefill path test

### CODE REVIEW

**Review by grimoire — APPROVED** [review]

File: lmdeploy/pytorch/backends/dlinfer/ascend/op_backend.py [hunk 1]
```diff
@@ -189,6 +190,14 @@ def get_total_slots():
                 cls.total_slots = cls.total_slots.view(block_num, block_size)
             return cls.total_slots
 
+        def get_fia_causal_mask():
+            device = step_context.block_offsets.device
+            mask = cls.fia_causal_masks.get(device)
+            if mask is None:
+                mask = torch.triu(torch.ones(2048, 2048, dtype=torch.int8, device=device), diagonal=1)
+                cls.fia_causal_masks[device] = mask
+            return mask
```
Comment (wanfengcxz) [inline lmdeploy/pytorch/backends/dlinfer/ascend/op_backend.py:199]: kernel不是按绝对位置直接访问，而是计算当前 Q/KV tile 相对于 causal 对角线的位置。

**Review by lvhan028 — APPROVED** [review]
